src/Model/Resolver/EntityUrl.php

```diff
@@ -139,7 +139,14 @@ public function resolve(
                 $collection = $this->productCollectionFactory->create()
                     ->addAttributeToFilter('status', ['eq' => Status::STATUS_ENABLED]);
                 $product = $collection->addIdFilter($id)->getFirstItem();
-                $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                $isInStock = false;
+
+                try {
+                    $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                    throw new NoSuchEntityException();
+                } catch (NoSuchEntityException $e) {
+                    // Ignoring error is safe
+                }
```

Comment:
What is going on here? So you throw the error manually to achieve what? This seem to me to work OK even without doing that `throw new NoSuchEntityException();`.